test/conformance/tests/sriov_operator.go

```diff
@@ -1701,9 +1701,20 @@ func waitForSRIOVStable() {
 	// TODO: find a better way to handle this scenario
 
 	time.Sleep(5 * time.Second)
+
+	eofErrorCount := 0
 	Eventually(func() bool {
 		res, err := cluster.SriovStable(operatorNamespace, clients)
-		Expect(err).ToNot(HaveOccurred())
+		if err != nil {
```

Comment:
This could be unrolled to avoid the extra nesting.
Ideally, the nested part is where the error is handled.
See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Also, in case of handled error I'd explicitly return false.
For example:
```
        if err == io.ErrUnexpectedEOF {
            eofErrorCount++
            Expect(eofErrorCount). // To be greater than 2
            return false
        }
        
        Expect(err).ToNot(HaveOccurred())
}

Reply:
Done